flink-models/flink-model-openai/pom.xml

```diff
@@ -128,4 +137,54 @@ under the License.
 			<url>https://repo1.maven.org/maven2</url>
 		</repository>
 	</repositories>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>shade-flink</id>
```

Comment:
I am curious. I see you are saying flink dependancies are provided i.e. picked up at runtime, but you are building a shaded Flink package. I was expecting something called shade-flink to be a shaded version of flink dependancies, but it does not look like it is as the Flink dependancies are not compile scoped.  

I was expecting either we pick up the flink dependancies at runtime or we bundle flink dependancies in a shaded package not both. Am I missing something?